From 8fb4c2d3282727b03e131e830fd667496668736b Mon Sep 17 00:00:00 2001
From: Zhiyi Zhang <zzhang@codeweavers.com>
Date: Thu, 6 Feb 2025 10:08:48 +0800
Subject: [PATCH 351/378] include: Fix PRTL_GENERIC_ALLOCATE_ROUTINE prototype.

The last parameter of PRTL_GENERIC_ALLOCATE_ROUTINE should be CLONG instead of LONG.
---
 include/ddk/ntddk.h | 2 +-
 include/ntdef.h     | 3 +++
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/ddk/ntddk.h b/include/ddk/ntddk.h
index e9ca903715d..bb084424987 100644
--- a/include/ddk/ntddk.h
+++ b/include/ddk/ntddk.h
@@ -169,7 +169,7 @@ typedef struct _RTL_SPLAY_LINKS
 struct _RTL_GENERIC_TABLE;
 
 typedef RTL_GENERIC_COMPARE_RESULTS (WINAPI *PRTL_GENERIC_COMPARE_ROUTINE)(struct _RTL_GENERIC_TABLE *, void *, void *);
-typedef void * (__WINE_ALLOC_SIZE(2) WINAPI *PRTL_GENERIC_ALLOCATE_ROUTINE)(struct _RTL_GENERIC_TABLE *, LONG);
+typedef void * (__WINE_ALLOC_SIZE(2) WINAPI *PRTL_GENERIC_ALLOCATE_ROUTINE)(struct _RTL_GENERIC_TABLE *, CLONG);
 typedef void (WINAPI *PRTL_GENERIC_FREE_ROUTINE)(struct _RTL_GENERIC_TABLE *Table, void *);
 
 typedef struct _RTL_GENERIC_TABLE
diff --git a/include/ntdef.h b/include/ntdef.h
index fc6d99835ed..f176bc20110 100644
--- a/include/ntdef.h
+++ b/include/ntdef.h
@@ -72,6 +72,9 @@ typedef unsigned int ULONG, *PULONG;
 #endif
 #endif
 
+typedef ULONG CLONG;
+typedef CLONG *PCLONG;
+
 typedef struct _RTL_BALANCED_NODE
 {
     union
-- 
2.51.0

